src/demo/demo.js

```diff
@@ -101,3 +55,80 @@ const getToken = function(onSuccess) {
   }
   request.send()
 }
+
+
+class SDK extends Component{
+  componentDidMount () {
+    this.initSDK(this.props.options)
+  }
+
+  componentWillReceiveProps({options}){
+    if (this.state.onfidoSdk){
+      this.state.onfidoSdk.setOptions(options)
+    }
+  }
+
+  componentWillUnmount () {
+    if (this.state.onfidoSdk) this.state.onfidoSdk.tearDown()
+  }
+
+  initSDK = (options)=> {
+    const onfidoSdk = Onfido.init(options)
+    this.setState({onfidoSdk})
+  }
+
+  shouldComponentUpdate () {
+    return false
+  }
+
+  render () {
+    return <div id="onfido-mount"></div>
+  }
+}
+
+class Demo extends Component{
+  constructor (props) {
+    super(props)
+    getToken((token)=> {
+      this.setState({token})
+    })
+  }
+
+  state = {
+    token: null,
+    isModalOpen: false
+  }
+
+  sdkOptions = ()=> ({
+    token: this.state.token,
+    useModal,
+    onComplete: () => {
+      /*callback for when */ console.log("everything is complete")
+    },
+    isModalOpen: this.state.isModalOpen,
+    language,
+    steps,
+    mobileFlow: !!options.link_id,
+    onModalRequestClose: () => {
+      this.setState({isModalOpen: false})
+    }
+  })
+
+  render = () => (
```

Comment:
[minor] you shouldn't really bind things that don't need binding, since there's a small performance impact to it
`render` should generally be safe to leave unbound, as it's a framework method

Reply:
yeah tbh I did it only for style in this case, I don't like `returns` cluttering the code :P But your point is valid, it's less optimised. Do you have numbers on performance impact of an arrow vs function? My guess was that is was a negligible impact.

Reply:
Oh, almost certainly negligible for most use-cases. [Someone created a JSPerf](https://stackoverflow.com/questions/17638305/why-is-bind-slower-than-a-closure) if you're interested, though - looks to be about 70% slower.